tests/e2e/vm_label_annotation_test.go

```diff
@@ -0,0 +1,237 @@
+
+	const (
+		labelName       = "os"
+		labelValue      = "ubuntu"
+		annotationName  = "test-annotation"
+		annotationValue = "true"
+	)
+
+	getPodName := func(namespace, resourceName string) string {
+		getPodCMD := "get pod --no-headers -o custom-columns=':metadata.name'"
+		subCMD := fmt.Sprintf("-n %s %s | grep %s", namespace, getPodCMD, resourceName)
+		podCMD := kubectl.RawCommand(subCMD, ShortWaitDuration)
+		podName := strings.TrimSuffix(podCMD.StdOut(), "\n")
+		return podName
+	}
+
+	getRecourseLabel := func(resourceType kc.Resource, resourceName string) *executor.CMDResult {
+		label := fmt.Sprintf("jsonpath='{.metadata.labels.%s}'", labelName)
+		stdout := kubectl.GetResource(resourceType, resourceName, kc.GetOptions{
+			Output:    label,
+			Namespace: conf.Namespace,
+		})
+		return stdout
+	}
+
+	getRecourseAnnotation := func(resourceType kc.Resource, resourceName string) *executor.CMDResult {
+		annotation := fmt.Sprintf("jsonpath='{.metadata.annotations.%s}'", annotationName)
+		stdout := kubectl.GetResource(resourceType, resourceName, kc.GetOptions{
+			Output:    annotation,
+			Namespace: conf.Namespace,
+		})
+		return stdout
+	}
+
+	BeforeAll(func() {
+		By("Apply image for vms")
+		ApplyFromFile(imageManifest)
+		WaitFromFile(imageManifest, PhaseReady, LongWaitDuration)
+	})
+	AfterAll(func() {
+		By("Delete all manifests")
+		files := make([]string, 0)
+		err := filepath.Walk(conf.VM.TestDataDir, func(path string, info fs.FileInfo, err error) error {
+			if err == nil && strings.HasSuffix(info.Name(), "yaml") {
+				files = append(files, path)
+			}
+			return nil
+		})
+		if err != nil || len(files) == 0 {
+			kubectl.Delete(imageManifest, kc.DeleteOptions{})
+			kubectl.Delete(conf.VM.TestDataDir, kc.DeleteOptions{})
+		} else {
+			for _, f := range files {
+				kubectl.Delete(f, kc.DeleteOptions{})
+			}
+		}
+	})
+
+	WaitVmStatus := func(name, phase string) {
+		GinkgoHelper()
+		WaitResource(kc.ResourceVM, name, "jsonpath={.status.phase}="+phase, LongWaitDuration)
+	}
+
+	Context("Label", func() {
+		var name string
+
+		vm, err := GetVMFromManifest(manifestVM)
+
+		BeforeAll(func() {
+			By("Apply manifest")
+			vm, err := GetVMFromManifest(manifestVM)
+			Expect(err).To(BeNil())
+			name = vm.Name
+			ApplyFromFile(manifestVM)
+			WaitVmStatus(name, VMStatusRunning)
+		})
+
+		AfterAll(func() {
+			By("Delete manifest")
+			kubectl.Delete(manifestVM, kc.DeleteOptions{})
+		})
+
+		Describe(fmt.Sprintf("Add label %s=%s", labelName, labelValue), func() {
+
+			It("Labeled", func() {
+				Expect(err).To(BeNil())
+				subCMD := fmt.Sprintf("-n %s label vm %s %s=%s", conf.Namespace, vm.Name, labelName, labelValue)
+
+				res := kubectl.RawCommand(subCMD, ShortWaitDuration)
+				Expect(res.Error()).NotTo(HaveOccurred())
+			})
+		})
+		Describe("Check label on resource", func() {
+			It("VM", func() {
+				res := getRecourseLabel(kc.ResourceVM, vm.Name)
+				Expect(res.StdOut()).To(Equal(labelValue))
```

Comment:
you are not handling errors.
you can do it like this
`Expect(res.Error()).NotTo(HaveOccurred(), "failed to get vm %s.\n%s", vm.Name, res.StdErr())`

Reply:
you can add error handling to your functions getResourceAnnotation, getRecourseLabel, etc